test/utils/clone-with.test.ts

```diff
@@ -0,0 +1,26 @@
+import * as Schema from '../../';
+
+describe('cloneWith()', function () {
+  it('should return a cloned object schema', () => {
+    const schema = Schema.object({
+      foo: Schema.string()
+    }, {
+      required: ['foo'],
+      additionalProperties: true
+    });
+
+    const cloned = Schema.cloneWith(schema, Schema.object({ bar: Schema.string() }, {
+      additionalProperties: true,
+      nullable: true
+    }));
+
+    expect(cloned).to.deep.equal({
```

Comment:
Does this result actually make sense? The object requires a `foo` property but has none? Why isn't `foo` still in the properties?